### ISSUE TITLE
odo push fails with "tar xf - -C /tmp/src --strip 1"

### ISSUE THREAD

**Issue by GeekArthur** [openshift/odo#2445]
/kind bug

## What versions of software are you using?

**Operating System:** 
MacOS

**Output of `odo version`:**
odo v1.0.1 (HEAD)

## How did you run odo exactly?
1. odo create nodejs nodejs
2. odo push

## Actual behavior
When I run `odo push`, it fails with the following errors:
```
Validation
 •  Checking component  ...
 ✓  Checking component [10ms]
Configuration changes
 ✓  Initializing component
 •  Creating component  ...
 ✓  Creating component [188ms]
Pushing to component cw-odonode123 of type local
 •  Checking files for pushing  ...
 ✓  Checking files for pushing [26ms]
 •  Waiting for component to start  ...
 ✓  Waiting for component to start [19s]
 •  Syncing files to the component  ...
E1205 16:22:36.774220   29664 occlient.go:2667] Command 'tar xf - -C /tmp/src --strip 1' in container failed.
E1205 16:22:36.774246   29664 occlient.go:2668] stdout: 
E1205 16:22:36.774255   29664 occlient.go:2669] stderr: tar: /tmp/src: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
E1205 16:22:36.774268   29664 occlient.go:2670] err: error while streaming command: command terminated with exit code 2
 ✗  Syncing files to the component [240ms]
 ✗  command terminated with exit code 2
```

## Expected behavior
`odo push` should have no errors and component should be built and pushed successfully to cluster

## Any logs, error output, etc?
Logs and errors are above. I talk with @kadel and his comments:
> [setup-s2i](https://github.com/openshift/odo-init-image/blob/91f6ee1fac349210cf6e34063d500656aebb25b3/s2i-setup#L22)  script that is executed as part of the [startup command](https://github.com/openshift/odo/blob/6ed4b57f3d644e536cf99b9951665f709360c481/pkg/occlient/templates.go#L92) of the container is making sure that   `/tmp/src`  is created. This is executed when the container is created. There might be some potential space for a race condition. It would mean that it took exceptionally long to execute setup-s2i script, and tar was created exceptionally quick. I haven't seen this before.

To resolve this, I think it’s good from the code flow perspective, definitely the issue should be related with timing issue, for example like @kadel said race condition makes `/tmp/src`  creation exceptionally long. Or probably other issues cause this but the root cause is `/tmp/src`  creation **end time** happens after tar **start time**. 

Possible solution is we add some code check before execute tar to wait for the existence of `/tmp/src` then execute the tar to tolerate the issue

**Comment by maysunfaisal** [comment]
Any updates on this one? :)

**Comment by GeekArthur** [comment]
@kadel We hit this issue frequently recently, can you please let someone take a look this issue? Thanks.

**Comment by girishramnani** [comment]
@GeekArthur we have moved this issue in the current sprint 🙂

**Comment by kadel** [comment]
It looks like this might get resolved in 1.0.2 by following changes. 

- https://github.com/openshift/odo-init-image/pull/43
- https://github.com/openshift/odo/pull/2427

We should verify that this is indeed the case.

**Comment by kadel** [comment]
We can't reproduce this anymore. It might get fixed.
I'm closing this for now.
@GeekArthur feel free to reopen if you still face this issue with the latest release.
/close
